test/GithubApi.Repositories.test.js

```diff
@@ -0,0 +1,81 @@
+const agent = require('superagent-promise')(require('superagent'), Promise);
+const statusCode = require('http-status-codes');
+const chai = require('chai');
+
+const { assert, expect } = chai;
+chai.use(require('chai-subset'));
+const md5 = require('md5');
+
+describe('Github Repositories Test', () => {
+  describe('• Consume the service https://api.github.com/users/aperdomob', () => {
+    const urlBase = 'https://api.github.com';
+    const githubUserName = 'aperdomob';
+    let userSession;
+
+    it('Validate userName, company and location', () => agent.get(`${urlBase}/users/${githubUserName}`)
+      .then((response) => {
+        userSession = response.body;
+
+        expect(response.status).to.equal(statusCode.OK);
+        expect(userSession.login).equal(githubUserName);
+        expect(userSession.name).equal('Alejandro Perdomo');
+        expect(userSession.company).equal('PSL');
+        expect(userSession.location).equal('Colombia');
+      }));
+
+    describe('• Get repository list', () => {
+      const expectedRepository = 'jasmine-awesome-report';
+      let repositories;
+      let repository;
+
+      it(`${expectedRepository} repository`, () => agent.get(userSession.repos_url)
```

Comment:
you can separate the API call from the assertion part by moving it into a `before` like this

```js
before(() =>
  agent.get(userSession.repos_url)
      .then((response) => {
          repositories = response.body;
          repository = repositories.find(repo => repo.name === expectedRepository);
    });
)

it(`${expectedRepository} repository`, () => {
    assert.exists(repository);
    expect(repository.full_name).to.equal('aperdomob/jasmine-awesome-report');
    expect(repository.private).to.equal(false);
    expect(repository.description).to.equal('An awesome html report for Jasmine');
});
```

that way we can separate the preconditions from the validations and keep the code more organized

Reply:
Hecho